src/python/include/Persistent_cohomology_interface.h

```diff
@@ -68,6 +68,44 @@ persistent_cohomology::Persistent_cohomology<FilteredComplex, persistent_cohomol
     return persistence;
   }
 
+  std::vector<std::vector<int>> cofaces_of_cubical_persistence_pairs() {
+
+    // Warning: this function is meant to be used with CubicalComplex only!!
+
+    auto pairs = persistent_cohomology::Persistent_cohomology<FilteredComplex,
+      persistent_cohomology::Field_Zp>::get_persistent_pairs();
+
+    // Gather all top-dimensional cells and store their simplex handles
+    std::vector<int> max_splx; for (auto splx : stptr_->top_dimensional_cells_range()){
+      max_splx.push_back(splx); 
+    }
+    // Sort these simplex handles and compute the ordering function
+    // This function allows to go directly from the simplex handle to the position of the corresponding top-dimensional cell in the input data 
+    std::unordered_map<int, int> order;  
+    //std::sort(max_splx.begin(), max_splx.end()); 
+    for (unsigned int i = 0; i < max_splx.size(); i++)  order.emplace(max_splx[i], i);
+
+    std::vector<std::vector<int>> persistence_pairs;
+    for (auto pair : pairs) {
+      int h = stptr_->dimension(get<0>(pair));
+      // Recursively get the top-dimensional cell / coface associated to the persistence generator
+      int face0 = stptr_->get_top_dimensional_coface_of_a_cell(get<0>(pair));
+      // Retrieve the index of the corresponding top-dimensional cell in the input data
+      int splx0 = order[face0];
+
+      int splx1 = -1;
+      if (isfinite(stptr_->filtration(get<1>(pair)))){
+      // Recursively get the top-dimensional cell / coface associated to the persistence generator
+      int face1 = stptr_->get_top_dimensional_coface_of_a_cell(get<1>(pair));
+      // Retrieve the index of the corresponding top-dimensional cell in the input data
+      splx1 = order[face1];
+      }
+      std::vector<int> vect{ h, splx0, splx1};
```

Comment:
It is a bit surprising to have a very different-looking output in the C++ part and the python part, but that's an implementation detail, it doesn't matter.

Reply:
Yes I agree. I did this because it was easier to only modify the code in the `.pyx` file. If necessary, I can change it, or I can open an issue later.